File: FlutterUI/flutter_ui/lib/core/services/validationService.dart
import 'package:flutter_ui/core/utilities/service.dart';

class ValidationService extends Service {
  /// If response has error returns true
  showErrors(Map<String, dynamic> jsonData) {
    if (jsonData["success"] as bool == true) {
      print("İşlem Başarılı");
      return false;
    }

    // Validation Control
    if (jsonData.containsKey("validationErrors")) {
      var validationErrors = jsonData["validationErrors"] as List;

      validationErrors.forEach((error) {
        print(error);
      });

      print(jsonData["message"]);
      return true;
    }

    // Security Control
    else if (jsonData.containsKey("securityError")) {
      print(jsonData["securityError"]);

      print(jsonData["message"]);
      return true;
    }

    // Transaction Control
    else if (jsonData.containsKey("transactionScopeError")) {
      print(jsonData["transactionScopeError"]);
      print(jsonData["message"]);

      return true;
    }

    // System Control
    else if (jsonData.containsKey("ErrorMessage")) {
      print(jsonData["ErrorMessage"]);

      return true;
    }

    // BusinessRule Control
    else if (jsonData.containsKey("message")) {
      print(jsonData["message"]);

      return true;
    }

    // Error
    else {
      print('Bir Şeyler Ters Gitti Beklenmedik Hata');

      return true;
    }
  }
}

File: AngularUI/angular-ui/src/app/core/services/validation.service.ts
import { ToastrService } from 'ngx-toastr';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class ValidationService {
  constructor(private toastrService: ToastrService) {}

  showErrors(responseError: any) {
    if (responseError.error != null) {
      // Validation Control
      if (responseError.error.data?.validationErrors != null) {
        responseError.error.data.validationErrors.forEach((error: any) => {
          this.toastrService.error(error, responseError.error.message);
        });

        return true;
      }

      // Security Control
      else if (responseError.error.data?.securityError != null) {
        this.toastrService.error(
          responseError.error.data.securityError,
          responseError.error.message
        );

        return true;
      }

      // Transaction Control
      else if (responseError.error.data?.transactionError != null) {
        this.toastrService.error(
          responseError.error.data.transactionError,
          responseError.error.message
        );

        return true;
      }

      // System Control
      else if (responseError.error.data?.errorMessage != null) {
        this.toastrService.error('', responseError.error.data.errorMessage);

        return true;
      } else if (responseError.error.message != null) {
        this.toastrService.error('', responseError.error.message);
      }

      // Error
      else {
        this.toastrService.error('Bir Şeyler Ters Gitti', 'Beklenmedik Hata');

        return true;
      }
    }

    return false;
  }
}
